src/lib/select/select.md

```diff
@@ -1,111 +1,166 @@
+this option. The content of the `<mat-option>` is what will be shown to the user.
 
 <!-- example(select-overview) -->
 
-### Simple select
+### Getting and setting the select value
 
-In your template, create an `mat-select` element. For each option you'd like in your select, add an
-`mat-option` tag. Note that you can disable items by adding the `disabled` boolean attribute or
-binding to it.
+The select component is set up as a custom value accessor, so you can manipulate the select's value
+using any of the form directives from the core `FormsModule` or `ReactiveFormsModule`: `ngModel`,
+`formControl`, etc. As with native `<select>`, `<mat-select>` also supports a `compareWith`
+function. (Additional information about using a custom `compareWith` function can be found in the
+[Angular forms documentation](https://angular.io/api/forms/SelectControlValueAccessor#caveat-option-selection)).
 
-*my-comp.html*
-```html
-<mat-form-field>
-  <mat-select placeholder="State">
-    <mat-option *ngFor="let state of states" [value]="state.code">{{ state.name }}</mat-option>
-  </mat-select>
-</mat-form-field>
-```
+<!-- example(select-form) -->
 
-### Getting and setting the select value
+`<mat-select>` also supports 2-way binding to the `value` property without the need for Angular
+forms.
 
-The select component is set up as a custom value accessor, so you can manipulate the select's value using
-any of the form directives from the core `FormsModule` or `ReactiveFormsModule`: `ngModel`, `formControl`, etc.
+<!-- example(select-value-binding) -->
 
-*my-comp.html*
-```html
-<mat-form-field>
-  <mat-select placeholder="State" [(ngModel)]="myState">
-    <mat-option *ngFor="let state of states" [value]="state.code">{{ state.name }}</mat-option>
-  </mat-select>
-</mat-form-field>
-```
+### Form field features
 
-*my-comp.ts*
-```ts
-class MyComp {
-  myState = 'AZ';
-  states = [{code: 'AL', name: 'Alabama'}...];
-}
-```
+There are a number of `<mat-form-field>` features that can be used with any `<mat-select>`. These
+include error messages, hint text, prefix & suffix, and theming. For additional information about
+these features, see the
+[form field documentation](https://material.angular.io/components/form-field/overview).
+
+<!-- example(select-hint-error) -->
+
+### Setting a static placeholder
+
+The placeholder is a text label displayed in the select trigger area when no value is selected. When
+a value is selected, the placeholder will float above the select trigger area. The placeholder can
+be specified either via a `placeholder` attribute on the `<mat-select>` or a `<mat-placeholder>`
+element in the same form field as the `<mat-select>`. The `<mat-form-field>` also has additional
+options for changing the behavior of the placeholder. For more information see the
+[form field placeholder documentation](https://material.angular.io/components/form-field/overview#floating-placeholder).
```

Comment:
I'd like to keep it here too since `placeholder` is an attribute that you place on the `<mat-select>` (in the future when we split the placeholder and label into separate concepts it can be completely handled by the form field).